### PR TITLE
feat/LIVE-4953 Rework the Debug menu and tools on LLM

### PR DESCRIPTION
**Pull request by juan-cortes**
### 📝 Description
Organize the screen and tools for the Debug menu tool available on LLM so that it's less messy, less intimidating to develop on, and more welcoming to keeping it up to date and useful. Not much functionality has changed so it should be easy to test, when QA'ing this, if a screen breaks it was probably breaking before so don't focus too much on that either, but track it so we can tackle it separately.

### ❓ Context

- **Impacted projects**: `ledger-live-mobile, ledger-live-common` 
- **Linked resource(s)**: `https://ledgerhq.atlassian.net/browse/LIVE-4953` 

### ✅ Checklist

- [x] **Test coverage** 
- [ ] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo
Demo available on slack as a 11 minute long video rant, sorry about that.

### 🚀 Expectations to reach
Play with the tool, improve it, use it.

### CODE REVIEW

**Review by grsoares21 — APPROVED** [review]
lgtm

File: apps/ledger-live-mobile/src/screens/Settings/Debug/Configuration/DebugEnv.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+  const envs = useMemo(() => {
+    const envs = getAllEnvs();
+    const definitions = envDefinitions;
+
+    Object.keys(envs).forEach(key => {
+      if (!(key in definitions)) {
+        const maybeDefinition = getDefinition(key);
+        if (maybeDefinition) {
+          definitions[key] = maybeDefinition.desc;
```
Comment (alexandremgo) [inline apps/ledger-live-mobile/src/screens/Settings/Debug/Configuration/DebugEnv.tsx:33]: why `maybe` though ? There is a possibility that `getDefinition` returns something not `null`, but that is actually not a description ?

File: apps/ledger-live-mobile/src/screens/Settings/Debug/Debugging/Store.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+import { useDispatch, useSelector } from "react-redux";
+import { Text } from "@ledgerhq/native-ui";
+import { useTheme } from "@react-navigation/native";
+import NavigationScrollView from "../../../../components/NavigationScrollView";
+import Button from "../../../../components/Button";
+import { SettingsActionTypes } from "../../../../actions/types";
+import { State } from "../../../../reducers/types";
+
+type Props = {
+  data: { [key: string]: unknown };
```
Comment (alexandremgo) [inline apps/ledger-live-mobile/src/screens/Settings/Debug/Debugging/Store.tsx:14]: You can do: `data: Partial<{ [key in keyof State]: unknown }>;`

**Review by alexandremgo — APPROVED** [review]
Except the small comments, LGTM 🔥

File: apps/ledger-live-mobile/src/screens/Settings/Debug/Configuration/DebugEnv.tsx (hunk 1 above)
Comment (juan-cortes) [reply apps/ledger-live-mobile/src/screens/Settings/Debug/Configuration/DebugEnv.tsx:33]: 🤔 I just wanted it to pass typecheck at that point, but indeed, the name doesn't make sense.
Feel free to commit if it bothers you

**Pull request merged**